v2/oauth/multiple-frontends-with-separate-backends.mdx

```diff
@@ -0,0 +1,286 @@
+
+Note that if the *frontends* and *backends* are in different *sub domains*, you don't need to use *OAuth* and can instead use [session sharing across sub domains](/docs/session/common-customizations/sessions/share-sessions-across-sub-domains).
+
+<img src="/img/oauth/multiple-frontend-domains-with-separate-backends.png" alt="Multiple Frontend Domains with separate Backends"/>
+
+The authentication flow will work in the following way:
+
+1. User accesses the `frontend` app
+2. The application `frontend` calls a login endpoint on the `backend` application.
+3. The `backend` application generates an `authorization` URL to the [**Authorization Service**](/docs/oauth/introduction#authorization-service) and redirects the user to it.
+4. The [**Authorization Service**](/docs/oauth/introduction#authorization-service) backend redirects the user to the login UI
+5. User completes the login attempt 
+6. The [**Authorization Service**](/docs/oauth/introduction#authorization-service) backend redirects the user to a `callback URL` that includes the **Authorization Code**.
+7. The **Authentication Code** gets sent to the application `backend` when the user agent accesses the callback URL
+8. The `backend` exchanges the **Authentication Code** for an [**OAuth2 Access Token**](/docs/oauth/introduction#oauth2-access-token)
+9. The `backend` saves the received token in a server session and sends it back to the `frontend` as a cookie.
+10. The `frontend` uses the new cookie to access protected resources from the `backend`.
+
+:::info
+This guide assumes that you already have setup and configured **SuperTokens** in your [**Authorization Service**](/docs/oauth/introduction#authorization-service). 
+
+For more information on how to do that please check our [quickstart guides](/docs/guides).
+:::
+
+## 1. Enable the OAuth2 features from the Dashboard
+
+You will first have to enable the **OAuth2** features from the **SuperTokens Dashboard**.
+1. Open the **SuperTokens Dashboard** on the main page
+2. Click on the **Enabled Paid Features** button
+3. Click on **Managed Service**
+4. Check the **OAuth 2.0** option
+5. Click *Save* 
+
+Now you should be able to use the OAuth2 recipes in your applications.
+
+## 2. Create the OAuth2 Clients
+
+<CoreInjector defaultValue="https://<your-core-identifier>.aws.supertokens.io" showTenantId={false}>
+
+<AppInfoForm
+    askForAppName
+    askForAPIDomain
+    askForWebsiteDomain
+>
+
+For each of your applications you will have to create a separate [**OAuth2 client**](/docs/oauth/introduction#client).
+This can be done by directly calling the **SuperTokens Core** API.
+
+```bash
+# You will have to run this for each one of your applications
+# Adjust the client_name and redirect_uri based on that
+curl -X POST ^{coreInjector_uri_without_quotes}/recipe/oauth2/admin/clients \
+  -H "Content-Type: application/json" \
```

Comment:
missing API key. Please see how other snippets to curl have the API key. Our component auto adds the api key if the user is logged into their dashboard

Reply:
resolved